### PR TITLE
Better messaging for componentDidCatch cross-origin errors

### LINKED ISSUE #10441: Cross-origin error passed to componentDidCatch incorrectly
tl;dr React is passing "A cross-origin error was thrown" to `componentDidCatch` when there are no cross-origin scripts.

See [this discussion thread](https://github.com/facebook/react/pull/10353#issuecomment-321528685) and [this repro case](https://github.com/tessin/tessin-mini).

I was able to confirm the behavior. A quick look at `onError` showed a null `event.error`.

### PR DESCRIPTION
**Pull request by bvaughn**
The prior message was a bit wordy and didn't cover all cases (like the one we discovered while researching #10441).

We can use the new URL to explain background info on the DEV-mode technique we're using _in addition to_ common fixes (eg `crossorigin` attribute for CDN scripts and Webpack `devtools` settings). Putting this information behind a link will allow us to more easily edit it in the future as common causes for this issue change.

Resolves #10441

**Note** I have not yet setup the `fb.me/react-crossorigin-error` shortcut or written the content that will live there. It's been a long day and I'm taking off. Technically we can merge this at any time (to unblock the RC) and just follow-up with the link soon. Or I'll add it on Monday if we're not in a hurry. 😄

### CODE REVIEW

File: src/renderers/shared/utils/ReactErrorUtils.js [hunk 1]
```diff
@@ -246,15 +246,9 @@ if (__DEV__) {
...
-            "A cross-origin error was thrown. React doesn't have access to " +
-              'the actual error because it catches errors using a global ' +
-              'error handler, in order to preserve the "Pause on exceptions" ' +
-              'behavior of the DevTools. This is only an issue in DEV-mode; ' +
-              'in production, React uses a normal try-catch statement.\n\n' +
-              'If you are using React from a CDN, ensure that the <script> tag ' +
-              'has a `crossorigin` attribute, and that it is served with the ' +
-              '`Access-Control-Allow-Origin: *` HTTP header. ' +
-              'See https://fb.me/react-cdn-crossorigin',
+            "A cross-origin error was thrown so React doesn't have access to " +
```
Comment (acdlite) [inline src/renderers/shared/utils/ReactErrorUtils.js]: Can we remove the "so" and make it two separate sentences?

Comment (bvaughn) [inline src/renderers/shared/utils/ReactErrorUtils.js]: Sure~

**Comment by bvaughn** [comment]
New docs page follow-up in #10457
